rustls/src/conn/unbuffered.rs

```diff
@@ -131,6 +131,18 @@ impl<Data> UnbufferedConnectionCommon<Data> {
                 .core
                 .common_state
                 .has_received_close_notify
+                && !self.emitted_peer_closed
```

Comment:
I'm starting to find the big chain of `if/else if`'s and their relatively complicated boolean expressions tricky to follow. It's quite dense.

I don't think it's necessary to clean up in this branch but I wonder if we could lift out some intermediate vars at the minimum so that the density is reduced a little bit.

Reply:
Yeah, agree. Even though I find the way the early data state is injected into `process_tls_records_common` a bit distasteful as a special case, I think actually it's a good direction to split the logic up a bit.

Reply:
Would suggest taking that out of this PR, though.